### PR TITLE
Normalization for general decoders

### PR DESCRIPTION
**Pull request by justjhong**
Uses full monte carlo on full inference

### CODE REVIEW

File: src/scvi_v2/_model.py [hunk 1]
```diff
@@ -411,39 +417,64 @@ def _compute_local_baseline_dists(self, batch: dict, mc_samples: int = 1000) ->
...
+
+                # figure out how to compute dists here
+                z = outputs["z"]
+                z = z.reshape((z.shape[0] // mc_samples_per_cell, mc_samples_per_cell, -1))
+                for cell_z in z:
+                    first_half_cell_z, second_half_cell_z = cell_z[:mc_samples], cell_z[mc_samples:]
+                    l2_dists_rows.append(
+                        jnp.sqrt(jnp.sum((first_half_cell_z - second_half_cell_z) ** 2, axis=1)).reshape((1, -1))
+                    )
+            l2_dists = jnp.concatenate(l2_dists_rows, axis=0)
```
Comment (PierreBoyeau) [inline src/scvi_v2/_model.py]: what is the inner for loop doing exactly?

**Review by PierreBoyeau — APPROVED** [review]

**Comment by justjhong** [comment]
@PierreBoyeau I think the order was wrong here so the changes here got lost. It should've been merged into adam_regs before adam_regs got merged into regs.
